spdlog/src/sink/rotating_file_sink.rs

```diff
@@ -63,6 +66,15 @@ pub enum RotationPolicy {
     },
     /// Rotating to a new log file at minute 0 of each hour.
     Hourly,
+    /// Rotating to a new log file after given duration (greater then {0, 0, 0}) is passed.
+    Duration {
+        /// Hours to the next rotation.. Range: [0, u32::MAX].
+        hours: u32,
+        /// Minutes to the next rotation. Range: [0, 59].
+        minutes: u32,
+        /// Seconds to the next rotation. Range: [0, 59].
+        seconds: u32,
+    },
```

Comment:
```suggestion
    /// Rotating to a new log file after given duration (greater then {0, 0, 0}) is passed.
    Period(Duration),
```

That's a little different than what I had in mind. [`std::time::Duration`](https://doc.rust-lang.org/stable/std/time/struct.Duration.html) should be more appropriate here as the parameter.

Considering a user wants to rotate every week, in the current way they have to write `Duration { hours: 7 * 24, minutes: 0, seconds: 0 }`, if we use `Duration`, just a [`Duration::from_weeks(1)`](https://doc.rust-lang.org/stable/std/time/struct.Duration.html#method.from_weeks) will work. Something like 1.5 hours can also be expressed as `Duration::from_hours(1) + Duration::from_mins(30)`.

And I prefer the name "Period" because we have already use the term in [`Logger::set_flush_period`](https://github.com/SpriteOvO/spdlog-rs/blob/813edaf99d910387bb5c8413f8022919009ccc92/spdlog/src/logger.rs#L308).